File: 30-days-of-code/php/day-15-linked-lists.php
<?php
class Node{
    public $data;
    public $next;
    function __construct($d)
    {
        $this->data = $d;
        $this->next = NULL;
    }
}
class Solution{

    function insert($head, $data){
        // Begin Solution
        //---------------------------------------------------------------------------------------
        if ($head == null) {
            $head = new Node($data);
        } else {
            $current = $head;
            while ($current->next != null) {
                $current = $current->next;
            }
            $current->next = new Node($data);
        }
        return $head;
        //---------------------------------------------------------------------------------------
        // End Solution
    }

function display($head){
        $start=$head;
        while($start){
            echo $start->data,' ';
            $start=$start->next;
        }
    }
}
$T=intval(fgets(STDIN));
$head=null;
$mylist=new Solution();
while($T-->0){
    $data=intval(fgets(STDIN));
    $head=$mylist->insert($head,$data);
}
$mylist->display($head);
?>

File: 30-days-of-code/cpp/day-15-linked-lists.cpp
#include <iostream>
#include <cstddef>
using namespace std;	
class Node {
    public:
        int data;
        Node *next;
        Node(int d){
            data=d;
            next=NULL;
        }
};

class Solution{
    public:

      Node* insert(Node *head,int data) {
        if (head == NULL) {
            head = new Node(data);
        } else {
            Node *current = head;
            while (current->next != NULL) {
                current = current->next;
            }
            current->next = new Node(data);
        }
        return head;
      }

      void display(Node *head)
}